airflow/api_fastapi/execution_api/datamodels/taskinstance.py

```diff
@@ -59,6 +59,8 @@ class TITerminalStatePayload(BaseModel):
     end_date: UtcDateTime
     """When the task completed executing"""
 
+    task_retries: int | None = None
```

Comment:
I think the fact that this shows up in the API makes it too confusing to client authors, so lets work out a way to remove this

Reply:
Agreed, it has been updated with "should_retry". That makes sense, or should we entirely decouple it even more having a seperate payload for failed state?